awscli/shorthand.py

```diff
@@ -149,9 +169,17 @@ def _parameter(self):
         # parameter = keyval *("," keyval)
         params = {}
         params.update(self._keyval())
+        last_index = self._index
         while self._index < len(self._input_value):
             self._expect(',', consume_whitespace=True)
-            params.update(self._keyval())
+            keyval = self._keyval()
+            key = list(keyval.keys())[0]
```

Comment:
It might be good to add a note why we are doing this. It was definitely not obvious first looking at it.